docs/tutorials/10_torch_qgan.ipynb

```diff
@@ -0,0 +1,752 @@
+      "  0.79287636  0.3463064  -0.6638512  -0.53881466  0.9884172  -0.63489616]\n",
+      "[Epoch 2/3] [Batch 2/8] [D loss: 0.665659] [G loss: 0.786490]\n",
+      "d loss  tensor(0.6790, grad_fn=<DivBackward0>)\n",
+      "g loss  tensor(0.7499, grad_fn=<BinaryCrossEntropyBackward0>)\n",
+      "generator params  [-0.61389863 -0.9106049   0.18855616  0.405817    0.85405123  0.3423761\n",
+      "  0.09741166 -0.590686    0.773005    0.3156932  -0.01166708 -0.0193572\n",
+      "  0.792719    0.34683084 -0.663374   -0.53786516  0.98978746 -0.6335248 ]\n",
+      "[Epoch 2/3] [Batch 3/8] [D loss: 0.678962] [G loss: 0.749932]\n",
+      "d loss  tensor(0.6552, grad_fn=<DivBackward0>)\n",
+      "g loss  tensor(0.7767, grad_fn=<BinaryCrossEntropyBackward0>)\n",
+      "generator params  [-0.61345506 -0.9112524   0.18917778  0.4068386   0.85277617  0.34386274\n",
+      "  0.098836   -0.59220314  0.7714497   0.31695262 -0.01130846 -0.01800152\n",
+      "  0.79259676  0.34740925 -0.662573   -0.5367618   0.991264   -0.63207805]\n",
+      "[Epoch 2/3] [Batch 4/8] [D loss: 0.655201] [G loss: 0.776672]\n",
+      "d loss  tensor(0.6750, grad_fn=<DivBackward0>)\n",
+      "g loss  tensor(0.7735, grad_fn=<BinaryCrossEntropyBackward0>)\n",
+      "generator params  [-0.6128028  -0.9120592   0.18982795  0.40805864  0.8514267   0.34519604\n",
+      "  0.10032255 -0.59354675  0.76976645  0.31770188 -0.01111185 -0.01713809\n",
+      "  0.7934951   0.34762403 -0.6617937  -0.5355773   0.9928516  -0.63058215]\n",
+      "[Epoch 2/3] [Batch 5/8] [D loss: 0.674995] [G loss: 0.773459]\n",
+      "d loss  tensor(0.6557, grad_fn=<DivBackward0>)\n",
+      "g loss  tensor(0.7773, grad_fn=<BinaryCrossEntropyBackward0>)\n",
+      "generator params  [-0.6120716  -0.9124608   0.19092356  0.4091579   0.849986    0.3465648\n",
+      "  0.10188712 -0.5948413   0.7681431   0.31863594 -0.01056974 -0.01660175\n",
+      "  0.79423827  0.34688115 -0.6612653  -0.5343422   0.99402374 -0.6291084 ]\n",
+      "[Epoch 2/3] [Batch 6/8] [D loss: 0.655654] [G loss: 0.777286]\n",
+      "d loss  tensor(0.6536, grad_fn=<DivBackward0>)\n",
+      "g loss  tensor(0.7904, grad_fn=<BinaryCrossEntropyBackward0>)\n",
+      "generator params  [-0.61130846 -0.9129811   0.19165756  0.41046792  0.848499    0.34807074\n",
+      "  0.10338581 -0.59606594  0.7665925   0.3199089  -0.01061776 -0.01568867\n",
+      "  0.79512256  0.34651956 -0.66038036 -0.5330397   0.9950392  -0.62763834]\n",
+      "[Epoch 2/3] [Batch 7/8] [D loss: 0.653618] [G loss: 0.790440]\n"
+     ]
+    }
+   ],
+   "source": [
+    "for epoch in range(n_epochs):\n",
+    "    for i, data in enumerate(dataloader):\n",
+    "\n",
+    "        # Adversarial ground truths\n",
+    "        valid = Variable(Tensor(data.size(0), 1).fill_(1.0), requires_grad=False)\n",
+    "        fake = Variable(Tensor(data.size(0), 1).fill_(0.0), requires_grad=False)\n",
```

Comment:
The use of `Variable`, while it works, is deprecated since torch 0.4. There are also easier ways to create tensors of zeros and ones :)

```suggestion
    "        valid = torch.ones((data.shape[0], 1), requires_grad=False)\n",
    "        fake = torch.zeros((data.shape[0], 1), requires_grad=False)\n",
```